dirs/dirs_test.go

```diff
@@ -38,6 +38,78 @@ var _ = Suite(&DirsTestSuite{})
+	snapHomeDirs = []string{"/home/homeDir1", "/home/homeDirs/homeDir1", "/home", "/home/homeDir2", "/home/homeTest/users"}
+	dataGlob = []string{"/home/homeDir1/*/snap", "/home/homeDirs/homeDir1/*/snap", "/home/*/snap", "/home/homeDir2/*/snap",
+		"/home/homeTest/users/*/snap"}
+	hiddenDataGlob = []string{"/home/homeDir1/*/.snap/data", "/home/homeDirs/homeDir1/*/.snap/data", "/home/*/.snap/data",
+		"/home/homeDir2/*/.snap/data", "/home/homeTest/users/*/.snap/data"}
+	getSnapHomeDirs = dirs.SnapHomeDirs()
+	c.Check(getSnapHomeDirs, DeepEquals, snapHomeDirs)
+	c.Check(getSnapHomeFromSet, DeepEquals, snapHomeDirs)
+	c.Check(dirs.DataHomeGlobs(nil), DeepEquals, dataGlob)
+	c.Check(dirs.DataHomeGlobs(&hidden), DeepEquals, hiddenDataGlob)
+
+	// Expected output should only return /ROOTDIR/home as a home directory
+	dirs.SetRootDir("/new/root")
+	getSnapHomeFromSet = dirs.SetSnapHomeDirs("")
+	snapHomeDirs = []string{"/new/root/home"}
+	dataGlob = []string{"/new/root/home/*/snap"}
+	hiddenDataGlob = []string{"/new/root/home/*/.snap/data"}
+	getSnapHomeDirs = dirs.SnapHomeDirs()
+	c.Check(getSnapHomeDirs, DeepEquals, snapHomeDirs)
+	c.Check(getSnapHomeFromSet, DeepEquals, snapHomeDirs)
+	c.Check(dirs.DataHomeGlobs(nil), DeepEquals, dataGlob)
+	c.Check(dirs.DataHomeGlobs(&hidden), DeepEquals, hiddenDataGlob)
+
+	// Expected output should append /ROOTDIR to each entry except if already present
+	getSnapHomeFromSet = dirs.SetSnapHomeDirs("/home/homeDir1,/new/root/home/homeDirs/homeDir1///,/home/homeDir2/,new/root/home/homeTest/users/")
+	snapHomeDirs = []string{"/new/root/home/homeDir1", "/new/root/home/homeDirs/homeDir1", "/new/root/home/homeDir2", "/new/root/home/homeTest/users", "/new/root/home"}
+	dataGlob = []string{"/new/root/home/homeDir1/*/snap", "/new/root/home/homeDirs/homeDir1/*/snap", "/new/root/home/homeDir2/*/snap",
+		"/new/root/home/homeTest/users/*/snap", "/new/root/home/*/snap"}
+	hiddenDataGlob = []string{"/new/root/home/homeDir1/*/.snap/data", "/new/root/home/homeDirs/homeDir1/*/.snap/data", "/new/root/home/homeDir2/*/.snap/data",
+		"/new/root/home/homeTest/users/*/.snap/data", "/new/root/home/*/.snap/data"}
+	getSnapHomeDirs = dirs.SnapHomeDirs()
+	c.Check(getSnapHomeDirs, DeepEquals, snapHomeDirs)
+	c.Check(getSnapHomeFromSet, DeepEquals, snapHomeDirs)
+	c.Check(dirs.DataHomeGlobs(nil), DeepEquals, dataGlob)
+	c.Check(dirs.DataHomeGlobs(&hidden), DeepEquals, hiddenDataGlob)
+
+	// setting a Root directory should reset the home directory
+	dirs.SetRootDir("/new")
+	c.Check(dirs.SnapHomeDirs(), DeepEquals, []string{"/new/home"})
+	c.Check(dirs.DataHomeGlobs(nil), DeepEquals, []string{"/new/home/*/snap"})
+	c.Check(dirs.DataHomeGlobs(&hidden), DeepEquals, []string{"/new/home/*/.snap/data"})
+	dirs.SetRootDir("/")
+	c.Check(dirs.SnapHomeDirs(), DeepEquals, []string{"/home"})
+	c.Check(dirs.DataHomeGlobs(nil), DeepEquals, []string{"/home/*/snap"})
+	c.Check(dirs.DataHomeGlobs(&hidden), DeepEquals, []string{"/home/*/.snap/data"})
+
+}
+
```

Comment:
I added some extra edge cases to help better test `dirs.SetSnapHomeDirs` however I also added extra checks for globbing expressions. I wasn't quite sure if I should add them to the same unit test but since most of the bulk of the operation was inside `dirs.SetSnapHomeDirs` I settled on this. 
Having a separate `TestDataHomeGlobs` would also make sense since it would make `TestSnapHomeDirs` more readable but at the cost of more code being duplicated. 
I would be interested to know what everyone thinks the better solution is.